code/Model/Indexer/Algoliacategories.php

```diff
@@ -156,18 +152,15 @@ protected function _processEvent(Mage_Index_Model_Event $event)
          * Category products are tracked separately. The specified categories are active. See _registerCatalogCategoryEvent().
          */
         if (!empty($data['catalogsearch_update_category_id'])) {
-            $updateCategoryIds = $data['catalogsearch_update_category_id'];
-            $updateCategoryIds = is_array($updateCategoryIds) ? $updateCategoryIds : [$updateCategoryIds];
-
-            foreach ($updateCategoryIds as $id) {
-                $categories = Mage::getModel('catalog/category')->getCategories($id);
-
-                foreach ($categories as $category) {
-                    $updateCategoryIds[] = $category->getId();
-                }
-            }
+            $this->indexer->reindexSpecificCategories($data['catalogsearch_update_category_id']);
+        }
 
-            $this->engine->rebuildCategoryIndex(null, $updateCategoryIds);
+        /**
+         * If we have added any new products to a category then we need t
+         * update that product in Algolia to reflect it.
```

Comment:
`If we have added any new products to a category then we need to update these products in Algolia indices`
Wouldn't it be better?